src/python/pants/engine/target.py

```diff
@@ -1667,6 +1655,27 @@ def __iter__(self) -> Iterator[Address]:
         return iter(self.dependencies)
 
 
+class SpecialCasedDependencies(AsyncStringSequenceField):
```

Comment:
> I don't feel like SpecialCased is very clear

I argue this is a feature, not a bug. There is very little unifying about this parent class, outside of a) these are like the normal dependencies field, and b) you are going to do something magical with the values, such that you didn't think it was appropriate to use the simpler, normal `Dependencies` field.

We don't know what the magic might be. It could be running the spiritual equivalent of `./pants package`, or it could be that you want to express this solely works with `files` targets, etc. 

My concern with `IndirectDependencies` is that it might be trying to generalize more than we can do in this situation. These are direct dependencies—rather than indirect—only that they have been split out from the normal `Dependencies` field for some reason the plugin author decided to do.

Wdyt?

Reply:
> We don't know what the magic might be. It could be running the spiritual equivalent of ./pants package, or it could be that you want to express this solely works with files targets, etc.

I said above that I would expect it to be better to describe what it *is* rather than what we *do* with it in the name, since as you describe here there are many different things we might want to do with it. I'm concerned that naming things based on how they're used leads to God Classes. I believe every other type of class in `target.py` follows the "what it *is*" naming scheme.

> These are direct dependencies—rather than indirect

Ok, I think I was mostly trying to go off of this description from the OP:
> or example, with `relocated_files()`, it's important that the original files targets are not included when we're hydrating sources for the transitive closure.

Could we consider a name that more accurately represents this distinction? The reason I thought about `IndirectDependenciesRequest` was because it chose to include the original `files()` target as well. Perhaps `IndirectDependencies` isn't appropriate for the field name itself -- maybe `WrappedDependencies`?